### PR TITLE
[fix] request_republish_group_task doesn't run on its own anymore

### PR DESCRIPTION
**Pull request by shipperizer**
@artsalliancemedia/thunderstorm

**## Non backwards compatibilities:**

* the datastore object now needs the group association model to be passed in, this is an effort to move all the group stuff into the datastore

### CODE REVIEW

File: thunderstorm_auth/datastore.py [hunk 1]
```diff
@@ -125,18 +152,19 @@ class SQLAlchemySessionAuthStore(SQLAlchemySessionStore, AuthStore):
...
 
-    def __init__(self, db_session, role_model, permission_model, association_model, bootstrap=False, cache=None):
+    def __init__(self, db_session, role_model, permission_model, association_model, group_model, bootstrap=False, cache=None):
         """
         Args:
             db_session (sqlalchemy session): database session
             role_model (sqlalchemy model): a role model class definition
             association_model (sqlalchemy model): a role-permission association model class definition
             permission_model (sqlalchemy model): a permission model class definition
+            group_model (list): A group model class definition (eg complex-group)
```
Comment (will-norris) [inline thunderstorm_auth/datastore.py]: `list`?

Comment (shipperizer) [reply thunderstorm_auth/datastore.py]: dang

File: thunderstorm_auth/datastore.py [hunk 2]
```diff
@@ -332,3 +360,49 @@ def delete_role_permission_association(self, role_uuid, permission_uuid, commit=
             self.commit()
 
         return (role_uuid, permission_uuid)
+
+    def have_group_associations(self):
```
Comment (will-norris) [inline thunderstorm_auth/datastore.py]: `def group_associations_exist(self):`

File: thunderstorm_auth/datastore.py [hunk 3]
```diff
@@ -332,3 +360,49 @@ def delete_role_permission_association(self, role_uuid, permission_uuid, commit=
...
+        """
+        Args:
+            group_uuid (uuid): primary identifier of a group
+
+        Returns:
+            query (sqlalchemy query object): query with all the group associations with that group uuid
+        """
+        return self.db_session.query(self.group_association_model).filter(self.group_association_model.group_uuid == group_uuid)
+
+    def get_groups_associations(self, group_uuids):
```
Comment (will-norris) [inline thunderstorm_auth/datastore.py]: `def get_associations_for_groups(self, group_uuids):`

**Pull request merged**